### ISSUE TITLE
Babel v6.0.2 systemjs transform plugin fails on simple file

### ISSUE THREAD

**Issue by pgbross** [babel/babel#2674]
Using Babel v6.0.2 with the following simple file:

```
export function myfunc(first, second){
  return first * second;
}
```

and a .babelrc with:

```
{
  "plugins": ["transform-es2015-modules-systemjs"]
}
```
fails with the error:

```
Error: example.js: We found a path that isn't a NodePath instance
```

For comparison, changing babelrc to use "transform-es2015-modules-commonjs" works correctly outputting  the expected transformation, so it looks like this is specific to the new systemjs transformer.

--philip

**Comment by zowers** [comment]
probably it's a duplicate of https://github.com/babel/babel/issues/2644

**Comment by sebmck** [comment]
Please post the full stack trace.

**Comment by pgbross** [comment]
Here is the stack trace:

```
ReferenceError: Unknown plugin "transform-es2015-modules-systemjs" specified in "/home/philip/tmp/test/.babelrc" at 0
    at /usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:179:17
    at Array.map (native)
    at Function.normalisePlugins (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:161:20)
    at OptionManager.mergeOptions (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:253:36)
    at OptionManager.addConfig (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:205:10)
    at OptionManager.findConfigs (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:346:16)
    at OptionManager.init (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/file/options/option-manager.js:391:12)
    at File.initOptions (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/file/index.js:191:75)
    at new File (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/file/index.js:122:22)
    at Pipeline.transform (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/pipeline.js:42:16)

```

hope that helps

**Comment by pgbross** [comment]
Here is the stack trace

```
Error: example.js: We found a path that isn't a NodePath instance
    at Function.get (/home/philip/tmp/test/node_modules/babel-traverse/lib/path/index.js:103:13)
    at TraversalContext.create (/home/philip/tmp/test/node_modules/babel-traverse/lib/context.js:73:30)
    at TraversalContext.visitSingle (/home/philip/tmp/test/node_modules/babel-traverse/lib/context.js:111:36)
    at TraversalContext.visit (/home/philip/tmp/test/node_modules/babel-traverse/lib/context.js:195:19)
    at Function.traverse.node (/home/philip/tmp/test/node_modules/babel-traverse/lib/index.js:139:17)
    at NodePath.visit (/home/philip/tmp/test/node_modules/babel-traverse/lib/path/context.js:106:22)
    at TraversalContext.visitQueue (/home/philip/tmp/test/node_modules/babel-traverse/lib/context.js:151:16)
    at TraversalContext.visitQueue (/home/philip/tmp/test/node_modules/babel-traverse/lib/context.js:157:21)
    at TraversalContext.visitMultiple (/home/philip/tmp/test/node_modules/babel-traverse/lib/context.js:106:17)
    at TraversalContext.visit (/home/philip/tmp/test/node_modules/babel-traverse/lib/context.js:193:19)
```

hope that helps

**Comment by schovi** [comment]
Same problem here. Only difference is it is inside webpack when requiring image `require("icons/main.png")`

Everything works fine before upgrading from babel 5 to 6 

```
Babel 6.0.12
Node 4.2.1
npm 3.3.10
webpack 1.12.2
```

.babelrc

```
{
  "presets": [
    "es2015"
  ]
}
```

stacktrace

**Comment by dashaus** [comment]
same problem here too. I had to return to babel 5.

**Comment by yzuniversal** [comment]
I have a similar issue, only that in my case the code is not so simple. Rather, there seem to be multiple factors here. Consts, regex, the if statement and the filter() method in the code below are all apparently necessary for the bug to occur. This code is roughly based on my actual code that didn't compile, only abstracted and shortened. I tried to remove everything (including es6 features) that didn't seem to cause the bug.

```
var test = true;
var testArrays = [
  ['test','tests','foobar'],
  ['test','testing','sjsjfdh'],
  ['something','nottest','whatever'],
];

function func() {
  const results = [];

  if (test) {
    const regex = /test/;

    testArrays.forEach(function(list) {
      results.push(list.filter(function(value) {
        return regex.test(value);
      }));
    });
  }

  return results;
}
```

I actually upgraded Node and npm in the meantime; the issue appears both on the most recent stable versions and on an older build. I'm using just `babel-cli` and `babel-preset-es2015` to test this, versions 6.0.14. The issue doesn't appear when I'm using Babel 5.

Here is the stack trace:
```
Error: main.js: We found a path that isn't a NodePath instance
    at Function.get (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/path/index.js:103:13)
    at TraversalContext.create (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/context.js:73:30)
    at TraversalContext.visitMultiple (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/context.js:102:25)
    at TraversalContext.visit (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/context.js:193:19)
    at Function.traverse.node (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/index.js:139:17)
    at traverse (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/index.js:61:12)
    at Function.traverse.hasType (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/index.js:188:3)
    at BlockScoping.wrapClosure (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/lib/index.js:491:47)
    at BlockScoping.run (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/lib/index.js:383:12)
    at PluginPass.BlockStatementProgram (/home/dominik/babel-test/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/lib/index.js:96:24)
```

I know that this looks very weird. Imagine my surprise when perfectly good code started to randomly cause an error during compilation and when there was no clear reason for this.

**Comment by zowers** [comment]
try `npm dedupe` as recommended in issue https://github.com/babel/babel/issues/2644

**Comment by sebmck** [comment]
"Fixed" via 24f5ef50b041a8659c904f0e8bdf88d5d0d77448. See the commit message and code comment for an explanation. Deduping is always going to be the most optimal fix.
